include/hybridmap/NeutralAtomScheduler.hpp

```diff
@@ -0,0 +1,160 @@
+   * @param qc Quantum circuit to schedule
+   * @param verbose If true, prints additional information
+   * @return SchedulerResults
+   */
+  SchedulerResults schedule(const qc::QuantumComputation&     qc,
+                            const std::map<HwQubit, HwQubit>& initHwPos,
+                            bool verbose, bool createAnimationCsv = false,
+                            fp shuttlingSpeedFactor = 1.0);
+
+  std::string getAnimationCsv() { return animationCsv; }
+  void        saveAnimationCsv(const std::string& filename) {
+    // save animation
+    std::ofstream file(filename);
+    file << animationCsv;
+    file.close();
+    // save architecture
+    auto filenameWithoutExtension =
+        filename.substr(0, filename.find_last_of('.'));
+    file.open(filenameWithoutExtension + "_architecture.csv");
+    file << animationArchitectureCsv;
+    file.close();
+  }
+
+  // Helper Print functions
+  static void printSchedulerResults(std::vector<fp>& totalExecutionTimes,
+                                    fp totalIdleTime, fp totalGateFidelities,
+                                    fp totalFidelities, uint32_t nCZs);
+  static void printTotalExecutionTimes(
+      std::vector<fp>&                            totalExecutionTimes,
+      std::vector<std::deque<std::pair<fp, fp>>>& blockedQubitsTimes);
+};
+
+class AnimationAtoms {
+  using axesId   = std::uint32_t;
+  using marginId = std::uint32_t;
+
+protected:
+  uint32_t                  colorSlm    = 0;
+  uint32_t                  colorAod    = 1;
+  uint32_t                  colorLocal  = 2;
+  [[maybe_unused]] uint32_t colorGlobal = 3;
+  uint32_t                  colorCz     = 4;
+
+  std::map<CoordIndex, HwQubit>        coordIdxToId;
+  std::map<HwQubit, std::pair<fp, fp>> idToCoord;
+  std::map<HwQubit, uint32_t>          axesIds;
+  std::map<HwQubit, uint32_t>          marginIds;
+  uint32_t                             axesIdCounter   = 0;
+  uint32_t                             marginIdCounter = 0;
+
+  axesId   addAxis(HwQubit id);
+  void     removeAxis(HwQubit id) { axesIds.erase(id); }
+  marginId addMargin(HwQubit id);
+  void     removeMargin(HwQubit id) { marginIds.erase(id); }
+
+public:
+  AnimationAtoms(const std::map<HwQubit, HwQubit>& initHwPos,
+                 const NeutralAtomArchitecture&    arch);
+
+  std::string        getInitString();
+  std::string        getEndString(fp endTime);
+  static std::string createCsvLine(fp startTime, HwQubit id, fp x, fp y,
+                                   uint32_t size = 1, uint32_t color = 0,
+                                   bool axes = false, axesId axId = 0,
+                                   bool margin = false, marginId marginId = 0,
+                                   fp marginSize = 0);
+  std::string createCsvOp(const std::unique_ptr<Operation>& op, fp startTime,
+                          fp endTime, const qc::NeutralAtomArchitecture& arch);
+};
+
```

Comment:
It might be worth to separate the animation code from the actual library code just to get a better separation of concerns. Maybe put all the related code into an `Animations.hpp` file.

Reply:
Good point. I added the animation later and didn't bother but in the end it was larger than expected. I'll try to move it in a separate file.

Reply:
Done, now the creation of the animation is in a Animations.hpp